src/integrations/IntegrationManagers.js

```diff
@@ -63,6 +77,42 @@ export class IntegrationManagers {
 
         if (apiUrl && uiUrl) {
             this._managers.push(new IntegrationManagerInstance(KIND_CONFIG, apiUrl, uiUrl));
+            this._primaryManager = null; // reset primary
+        }
+    }
+
+    async _setupHomeserverManagers() {
+        try {
+            console.log("Updating homeserver-configured integration managers...");
+            const homeserverDomain = MatrixClientPeg.getHomeserverName();
+            const discoveryResponse = await AutoDiscovery.getRawClientConfig(homeserverDomain);
```

Comment:
This will do it's own .well-known request, right?
Is there no way here to re-use the response of the .well-known response request to determine the homeserver?